src/raftstore/store/fsm/apply.rs

```diff
@@ -1841,18 +1792,26 @@ impl ApplyDelegate {
                 "source_region_id" => source_region_id
             );
 
-            let mailbox = ctx.router.mailbox(self.region_id()).unwrap();
-            let ready_to_merge = Arc::new(AtomicU64::new(0));
-            let msg = Msg::CatchUpLogs(CatchUpLogs {
-                target_mailbox: mailbox,
-                merge: merge.to_owned(),
-                ready_to_merge: ready_to_merge.clone(),
-            });
-            ctx.router.schedule_task(source_region_id, msg);
-            return Ok((
-                AdminResponse::default(),
-                ApplyResult::WaitMergeSource(ready_to_merge),
-            ));
+            if let Some(mailbox) = ctx.router.mailbox(self.region_id()) {
+                let logs_up_to_date = Arc::new(AtomicU64::new(0));
+                let msg = Msg::CatchUpLogs(CatchUpLogs {
+                    target_mailbox: mailbox,
+                    merge: merge.to_owned(),
+                    logs_up_to_date: logs_up_to_date.clone(),
+                });
+                ctx.router.schedule_task(source_region_id, msg);
+                return Ok((
+                    AdminResponse::default(),
+                    ApplyResult::WaitMergeSource(logs_up_to_date),
+                ));
+            } else {
```

Comment:
Is it necessary? /cc @BusyJay

Reply:
Yes, it does fail in the test. It is due to the batch system close mailboxes before joining all worker threads when shutting down, I will fix it in other PR. 
Anyway, we think it is better to check it here.

Reply:
The check here can't solve the problem. Because `mailbox()` will not lock the router, hence there still be a chance that the task is dropped inside `schedule_task`.

Reply:
seems it is okay to drop the task. The check here is to avoid unwrapping a `None`